<?xml version="1.0"?>
<!--
Copyright (C) 2024 Bellande Robotics Sensors Research Innovation Center, Ronaldson Bellande

Licensed under the Apache License, Version 2.0 (the "License"); you may not
use this file except in compliance with the License. You may obtain a copy of
the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
License for the specific language governing permissions and limitations under
the License.
-->
<launch>
  <!-- Launch the point cloud source (adjust parameters as needed) -->
  <node name="pointcloud_publisher" pkg="ros_web_api_bellande_3d_computer_vision"
    type="pointcloud_pradiction_node"
    output="screen">
    <param name="frame_id" value="base_link" />
  </node>

  <!-- Launch the prediction node -->
  <node name="pointcloud_prediction_node" pkg="ros_web_api_bellande_3d_computer_vision"
    type="pointcloud_prediction_node.py" output="screen">
    <remap from="input_pointcloud" to="/pointcloud_topic" />
  </node>

  <!-- Launch RViz for visualization -->
  <node name="rviz" pkg="rviz" type="rviz"
    args="-d $(find ros_web_api_bellande_3d_computer_vision)/rviz/pointcloud_visualization.rviz" />

</launch>
